library/src/scala/IArray.scala

```diff
@@ -260,59 +248,125 @@ object IArray:
+    def reverseIterator: Iterator[T] = genericArrayOps(arr).reverseIterator
+    def search[U >: T](elem: U)(using Ordering[U]): Searching.SearchResult = arr.toSeq.search(elem)
+    def search[U >: T](elem: U, from: Int, to: Int)(using Ordering[U]): Searching.SearchResult = arr.toSeq.search(elem, from, to)
+    def sizeCompare(that: IArray[Any]): Int = arr.toSeq.sizeCompare(that)
+    def sizeCompare(that: Iterable[_]): Int = arr.toSeq.sizeCompare(that)
+    def sizeCompare(otherSize: Int): Int = genericArrayOps(arr).sizeCompare(otherSize)
+    def sliding(size: Int, step: Int = 1): Iterator[IArray[T]] = genericArrayOps(arr).sliding(size, step)
+    def stepper[S <: Stepper[_]](using StepperShape[T, S]): S = genericArrayOps(arr).stepper[S]
+    def tails: Iterator[IArray[T]] = genericArrayOps(arr).tails
+    def tapEach[U](f: (T) => U): IArray[T] =
+      arr.toSeq.foreach(f)
+      arr
+    def unzip[T1, T2](using asPair: T => (T1, T2), ct1: ClassTag[T1], ct2: ClassTag[T2]): (IArray[T1], IArray[T2]) = genericArrayOps(arr).unzip
+    def unzip3[T1, T2, T3](using asTriple: T => (T1, T2, T3), ct1: ClassTag[T1], ct2: ClassTag[T2], ct3: ClassTag[T3]): (IArray[T1], IArray[T2], IArray[T3]) = genericArrayOps(arr).unzip3
+    def updated[U >: T: ClassTag](index: Int, elem: U): IArray[U] = genericArrayOps(arr).updated(index, elem)
+    def view: SeqView[T] = genericArrayOps(arr).view
+    def withFilter(p: T => Boolean): WithFilter[T] = new WithFilter(p, arr)
+    def zip[U](that: IArray[U]): IArray[(T, U)] = genericArrayOps(arr).zip(that)
+    def zip[U](that: IterableOnce[U]): IArray[(T, U)] = genericArrayOps(arr).zip(that)
+    def zipAll[T1 >: T, U](that: IArray[U], thisElem: T1, thatElem: U): IArray[(T1, U)] = genericArrayOps(arr).zipAll(that, thisElem, thatElem)
+    def zipAll[T1 >: T, U](that: Iterable[U], thisElem: T1, thatElem: U): IArray[(T1, U)] = genericArrayOps(arr).zipAll(that, thisElem, thatElem)
+    def zipWithIndex: IArray[(T, Int)] = genericArrayOps(arr).zipWithIndex
+  end extension
+
+  extension [T](arr: IArray[T])
+    def transpose[U](implicit asArray: T => IArray[U]): IArray[IArray[U]] =
+      genericArrayOps(arr).transpose(using asArray.asInstanceOf[T => Array[U]])
+
+  extension [T, U >: T: ClassTag](prefix: IterableOnce[T])
+    def ++:(arr: IArray[U]): IArray[U] = genericArrayOps(arr).prependedAll(prefix)
+
+  extension [T, U >: T: ClassTag](prefix: IArray[T])
+    def ++:(arr: IArray[U]): IArray[U] = genericArrayOps(arr).prependedAll(prefix)
+
+  extension [T, U >: T: ClassTag](x: T)
+    def +:(arr: IArray[U]): IArray[U] = genericArrayOps(arr).prepended(x)
 
   /** Conversion from IArray to immutable.ArraySeq */
-  extension [T](arr: IArray[T]) def toSeq: immutable.ArraySeq[T] =
-    immutable.ArraySeq.unsafeWrapArray(arr.asInstanceOf[Array[T]])
+  given genericWrapIArray[T](using DummyImplicit): Conversion[IArray[T], scala.collection.immutable.ArraySeq[T]] =
```

Comment:
Why the `DummyImplicit`s, here and below?

Reply:
If I don't I get an error at call site

```scala
[error] -- [E007] Type Mismatch Error: library/src/scala/runtime/TupleXXL.scala:26:18 
[error] 26 |        while i < es.length do
[error]    |                  ^^
[error]    |Found:    (TupleXXL.this.es : IArray[Object])
[error]    |Required: ?{ length: ? }
[error]    |Note that implicit extension methods cannot be applied because they are ambiguous;
[error]    |both object IArray in package scala and given instance genericWrapIArray in object IArray provide an extension method `length` on (TupleXXL.this.es : IArray[Object])
```

Maybe they should be placed in Predef as are the ones for Arrays. But adding them to `stdLibPatches/Predef.scala` crashed the compiler.

Reply:
Avoided this with `implicit def`